app/views/file_upload.py

```diff
@@ -0,0 +1,52 @@
+from django.shortcuts import render, redirect
+
+from app.models import ColumnMetadata, TableMetadata
+
+
+def file_upload(request, table_id):
+    if request.method == "GET":
+        return render_file_upload_page(request, table_id, "")
+
+    upload_error = extract_columns_and_get_error(table_id, request.FILES)
+    if upload_error:
+        return render_file_upload_page(request, table_id, upload_error)
+
+    return redirect("/edit-table-columns/" + str(table_id))
+
+
+def render_file_upload_page(request, table_id, upload_error):
+    table_metadata = TableMetadata.objects.get(pk=table_id)
+    file_upload_data = {"table_name": table_metadata.name, "upload_error": upload_error}
+    return render(request, "file-upload.html", file_upload_data)
+
+
+def extract_columns_and_get_error(table_id, files):
+    # File is missing
+    uploaded_file = files.get("uploaded_file", None)
+    if not uploaded_file:
+        return "The file is missing!"
```

Comment:
```suggestion
        raise StopUpload(connection_reset=True)
```

Use a `raise` rather than `return` (same as below). `connection_reset=True` says to stop the upload and restart the connection.

Reply:
I think the StopUpload exception is required when you create a custom FileUploadHandler. I am not doing that, but it might be a create way to propagate the error up